articles/migrations/guides/extensibility-node12.md

```diff
@@ -0,0 +1,221 @@
+
+To update it, you need to save the extension's settings (no changes are necessary). To do so, after switching the runtime to `Node 12`, you need to open the extension's settings in the extensions dashboard (gear icon) and hit `Save`. After that, the extensions gallery will update the `PUBLIC_WT_URL` secret accordingly based on the selected runtime.
+
+## How to ensure a stable migration
+
+As part of the process of introducing Node 12 in our Webtask runtime, we ran a number of tests to determine which modules are not forward-compatible from Node 8 to 12. Most customers _should_ be able to upgrade to Node 12 without any issues.
+
+With that said, before you migrate, we highly recommend testing all of your:
+
+* Rules
+* Hooks
+* Custom Database Connections/Scripts
+* Custom Social Connections
+* Extensions
+
+Furthermore, we recommend that the testing be done in your development tenant and migrating your production tenant only if you see no issues in development. 
+
+You can query the Management API for your Rules, Custom Database scripts, and Custom Social Connections. This will make it easier for you to move items from your production tenant to development tenant for testing purposes.
+
+Please see our documentation on the [Connections](/api/management/v2#!/Connections) and [Rules](/api/management/v2#!/Rules/get_rules) endpoints for additional information on this process.
+
+When using the [Connections](/api/management/v2#!/Connections) endpoints in the Management API, Custom Database Scripts can be retrieved or updated using `options.customScripts`.
+
+Similarly, you can find Custom Social Connections in `options.scripts.fetchUserProfile`.
+
+You will need to manually copy over any Hooks-related code that you use since they cannot be accessed via the Management API.
+
+## Affected modules
+
+If you are using the following built-in modules (that is, modules that you did not explicitly require), please be aware that some versions were updated to work with Node 12. The following table summarizes the changes.
+
+| Module name | Old version | New version |
+| - | - | - |
+| azure-storage | ~0.4.1 | ~2.2.1 |
+| couchbase | ~1.2.1 | ~2.3.5 |
+| jsonwebtoken | ~0.4.1^* | ~7.4.1 (w/ compatibility shim) |
+| knex | ~0.6.3 | ~0.13.0 |
+| mongo-getdb | ~1.2.0^* | ^2.2.0 |
+| mongodb | ~1.3.15^* | ^2.2.0 |
+| mysql | 2.0.0-alpha8^* | ^2.0.0 |
+| node-cassandra-cql | ^0.4.4 | ^0.5.0 |
+| request | ~2.27.0 | ~2.81.0 |
+| pg | ^4.3.0^* | ^4.5.7 |
+| bcrypt | ~0.8.5^* | 1.0.3 |
+| xml2json | ~0.10.0^* | ~0.11.2 |
+
+^* These versions are no longer supported due to incompatibility with Node 12.
+
+### Pinned modules
+
+If you have manually pinned modules, you may need to manually update them so that your code runs with Node 12.
+
+For example, you must change
+
+`var mysql = require(‘mysql@2.0.0-alpha8’);`
+
+to
+
+`var mysql = require(‘mysql’);`
+
+or, if the module must be pinned to a specific version:
+
+`var mysql = require(‘mysql@2.0.0’);`
```

Comment:
Since MySQL didn't change, lets use bcrypt example.
```suggestion
`var bcrypt = require(‘bcrypt@1.0.3’);`

to

`var bcrypt = require(‘bcrypt’);`

or, if the module must be pinned to a specific version:

`var bcrypt = require(‘bcrypt@3.0.8’);`
```